discount/src/main/java/com/kyriba/conference/discount/service/NoSuchDiscountException.java

```diff
@@ -0,0 +1,21 @@
+package com.kyriba.conference.discount.service;
+
+import java.util.NoSuchElementException;
+
+/**
+ * @author Igor Lizura
+ */
+public class NoSuchDiscountException extends NoSuchElementException {
```

Comment:
It is beter to use RuntimeException as parent to avoid side effects

Reply:
Please, explain.
NoSuchElementException - Thrown by various accessor methods to indicate that the element being requested does not exist. This is exactly what I want.

Reply:
This is an exception from Java Core and is actively used by different libraries. And they could catch this exception and process it somehow.